app/app.rb

```diff
@@ -0,0 +1,79 @@
+ENV['RACK_ENV'] ||= 'development'
+
+require 'sinatra/base'
+require './app/data_mapper_setup'
+require 'bcrypt'
+require 'sinatra/flash'
+
+class Chitter < Sinatra::Base
+  use Rack::MethodOverride
+  enable :sessions
+  set :session_secret, 'super secret'
+  register Sinatra::Flash
+
+	get '/links' do
+		@links=Message.all
+		erb(:'links/index')
+	end
+
+  post '/links' do
+  link = Message.new(message: params[:message])
+  link.save
+  redirect to('/links')
+end
+
+  get '/' do
+    redirect '/sign_up'
+  end
+
+  post '/links/new' do
+    erb(:'links/new')
+  end
+
+  get '/sign_up' do
+    @user = User.new
+    erb(:'users/sign_up')
+  end
+
+  post '/users' do
+  @user = User.new(email: params[:email],
+                  password: params[:password],
+                  password_confirmation: params[:password_confirmation])
+  if @user.save
+    session[:user_id] = @user.id
+    redirect to('/links')
+  else
+    flash.now[:errors] = @user.errors.full_messages
+    erb :'users/sign_up'
+   end
+  end
+
+  get '/sessions/new' do
+    erb :'sessions/new'
+  end
+
+  delete '/sessions' do
+    session[:user_id] = nil
+    flash.keep[:notice] = 'goodbye!'
+    redirect to '/links'
+end
```

Comment:
Lint/BlockAlignment: `end` at 59, 0 is not aligned with `delete '/sessions' do` at 55, 2.